sdk/lib/rtl/heap.c

```diff
@@ -353,21 +353,23 @@ RtlpInsertFreeBlockHelper(PHEAP Heap,
     Heap->FreeHints[HintIndex] = &FreeEntry->FreeList;
 }
 
-VOID NTAPI
+LONG NTAPI
```

Comment:
Looking again at my previous questions, it happens that PR #4915 chooses instead to return a pointer to the (updated) free entry, see https://github.com/reactos/reactos/pull/4915/files#diff-b4b704c9538d6bc32a5a1e458b2c9b06a81dc9723c2c17902d7daa8bf054adb7L356
This looks to me more "robust", in the sense that we wouldn't have to care about the type of the FreeSize variable there (and could be useful if the caller of this function would, one day, need to access extra information about the last newly created free block that has been inserted).

Reply:
From Above:
> It no longer fixed Process Hacker for me. I have now updated it with info from @ThFabba's PR https://github.com/reactos/reactos/pull/4915 and it now fixes Process Hacker again. I only used the most basic part of his PR. There is still much to gain with what he has in his PR.

My intent was not to "copy" Thomas Faber's work, but to only "borrow" the minimum from it for now. I agree that what he has is more "robust", but this should be left to him and the PR #4915. Thanks.

Reply:
OK then, borrowing the minimum from his PR, you could make that function actually return the pointer to the newest FreeEntry (as he did in his PR); this would also remove the need to add a new "LONG FreeSize;" in that function (thus, reduced diff).
Then, you could use the `FreeEntry->Size` where you need the "FreeSize".

Reply:
@HBelusca, You make a good point here. BUT, this PR has been open 9+ months and my Thomas Faber's additions where done 5+ months ago. It now has 3 approvals and I had to "beg" for these and wait a long time. If I make your suggested changes, I fear that I would risk having to start all over with obtaining approvals. Are you OK with proceeding with this as it is and we can come back with another PR later to improve it? Thanks.

Reply:
> could be useful if the caller of this function would, one day, need to access extra information about the last newly created free block

That is a stupid argument. Don't fix what's not broken.

Reply:
Now following what Thomas Faber proposed here in his PR #4915.